lib/src/signed_video_h26x_sign.c

```diff
@@ -98,9 +98,9 @@ add_payload_to_buffer(signed_video_t *self, uint8_t *payload, uint8_t *payload_s
     return;
   }
 
-  self->payload_buffer[2 * self->payload_buffer_idx] = payload;
-  self->payload_buffer[2 * self->payload_buffer_idx + 1] = payload_signature_ptr;
-  self->payload_buffer_idx += 1;
+  self->payload_buffer[self->payload_buffer_idx] = payload;
+  self->payload_buffer[self->payload_buffer_idx + 1] = payload_signature_ptr;
+  self->payload_buffer_idx += 2;
```

Comment:
Now when the behavior of `payload_buffer_idx` has changed you need to adjust the sanity check on line 95 above